src/extensions/interpolation.jl

```diff
@@ -4,19 +4,24 @@
 # from within the `@cm_str` macro otherwise the expressions won't have "time"
 # to actually evaluate.
 
+# Captures an interpolated Julia expression and its position in the string
+struct JuliaExpression <: AbstractInline
+    pos::Int
+    ex
+    JuliaExpression(pos, ex) = new(pos, ex)
+end
```

Comment:
Should there be the `write_*` methods defined on `JuliaExpression` since it is an `AbstractInline`? If any values of those types were to leak into a final AST (if they can even?) what gets printed at the moment?

Reply:
Ah, this is a fair point, I forgot about he `write_*` methods. In principle, as long as you use the `cm""` macro, I don't think it should be possible for `JuliaExpression` to leak, but they would end up in the AST if you construct your own parser with a `JuliaInterpolationRule`.

I think it would be reasonable to have the methods defined, and I guess they should do what the Markdown writer does right now and just print the expression with `$(...)` around it?

Reply:
> I guess they should do what the Markdown writer does right now and just print the expression with `$(...)` around it?

Sounds reasonable.